tests/test_feed.py

```diff
@@ -0,0 +1,167 @@
+def feed_response_missing_package(httpserver):
+  httpserver.expect_ordered_request(
+      '/api/v3/feeds/files/200102030405',
+      method='GET',
+      headers={'X-Apikey': 'dummy_api_key'}
+  ).respond_with_json({
+      'error': {
+          'code': 'NotFoundError'}}, status=404)
+
+  httpserver.expect_ordered_request(
+      '/api/v3/feeds/files/200102030406',
+      method='GET',
+      headers={'X-Apikey': 'dummy_api_key'}
+  ).respond_with_data(
+      bz2.compress(b'{\"type\": \"file\", \"id\": \"dummy_file_id_4\"}'))
+
+
+def test_interface(httpserver):
+  """Tests feed's interface. Checks if the object is an actual iterator."""
+  with new_client(httpserver) as client:
+    feed = client.feed(FeedType.FILES)
+
+  assert isinstance(feed, abc.Iterator)
+  assert isinstance(feed, abc.AsyncIterator)
+
+
+def test_next(httpserver, feed_response):
+  """Tests feed's next."""
+  with new_client(httpserver) as client:
+    feed = client.feed(FeedType.FILES, cursor='200102030405')
+
+    obj = next(feed)
+    assert obj.type == 'file'
+    assert obj.id == 'dummy_file_id_1'
+
+    obj = next(feed)
+    assert obj.type == 'file'
+    assert obj.id == 'dummy_file_id_2'
+
+    obj = next(feed)
+    assert obj.type == 'file'
+    assert obj.id == 'dummy_file_id_3'
+
+    # Iteration must start right where the last 'next' left
+    for obj in feed:
+      assert obj.type == 'file'
+      assert obj.id == 'dummy_file_id_4'
+      assert feed._count == 4
+      break  # Exit loop as the feed iteration doesn't stop
+
+
+@pytest.mark.asyncio
+async def test_anext(httpserver, feed_response):
+  """Tests feed's async next."""
+  async with new_client(httpserver) as client:
+    feed = client.feed(FeedType.FILES, cursor='200102030405')
+    obj = await feed.__anext__()
+    assert obj.type == 'file'
+    assert obj.id == 'dummy_file_id_1'
+
+    obj = await feed.__anext__()
+    assert obj.type == 'file'
+    assert obj.id == 'dummy_file_id_2'
+
+    obj = await feed.__anext__()
+    assert obj.type == 'file'
+    assert obj.id == 'dummy_file_id_3'
+
+    # Iteration must start right where the last 'next' left
+    async for obj in feed:
+      assert obj.type == 'file'
+      assert obj.id == 'dummy_file_id_4'
+      assert feed._count == 4
+      break  # Exit loop as the feed iteration doesn't stop
+
+
+@pytest.mark.parametrize("test_tolerance", [0, 1, 2])
+def test_tolerance(httpserver, feed_response_missing_package, test_tolerance):
+  """Tests feed's tolerance to missing packages."""
+  with new_client(httpserver) as client:
+    feed = client.feed(FeedType.FILES, cursor='200102030405')
+    feed._missing_batches_tolerancy = test_tolerance
+
+    if feed._missing_batches_tolerancy:
+      obj = next(feed)
+      assert obj.type == 'file'
+      assert obj.id == 'dummy_file_id_4'
+      assert feed._count == 1
+    else:
+      with pytest.raises(APIError) as e_info:
+        obj = next(feed)
+      assert e_info.value.args[0] == 'NotFoundError'
```

Comment:
I'd add an extra step to this test, the first (or two first) requested elements will be skipped correctly but after 1 (or two) consecutive missing batches an error should be raised.

Reply:
Perfect. Please check the new commit and let me know if that is what you had in mind.